ladybug/windrose.py

```diff
@@ -0,0 +1,761 @@
+            min_bar_radius = self.zeros_per_bin / max_bar_num * max_bar_radius
+            ytick_num_mesh = max_bar_num / self.frequency_maximum * ytick_num
+
+            # Update yticks
+            zero_ytick_num_mesh = self._ytick_zero_inc
+            ytick_num_mesh -= zero_ytick_num_mesh
+
+            # Compute the array for calm rose
+            zero_data = [[0] for _ in self.histogram_data]
+            zero_data_stacked = [[0] for _ in self.histogram_data]
+            zero_mesh_array, zero_color_array = WindRose._compute_colored_mesh_array(
+                zero_data, zero_data_stacked, self.bin_vectors, zero_ytick_num_mesh,
+                0, min_bar_radius, show_stack=False)
+
+        # Calculate regular mesh
+        if self.show_stack:
+            histogram_data_stacked = WindRose._histogram_data_stacked(
+                self.histogram_data, ytick_num)
+        else:
+            histogram_data_stacked = [[sum(h) / len(h)]  if len(h) > 0 else [0]
+                                      for h in self.histogram_data]
+        mesh_array, color_array = WindRose._compute_colored_mesh_array(
+            self.histogram_data, histogram_data_stacked, self.bin_vectors,
+            ytick_num_mesh, min_bar_radius, max_bar_radius, self.show_stack)
+
+        mesh_array += zero_mesh_array
+        color_array += zero_color_array
+
+        mesh = Mesh2D.from_face_vertices(mesh_array, purge=True)
+        mesh.colors = color_array
+
+        return mesh.scale(self.wind_radius).move(self.base_point)
+
+    @property
+    def orientation_lines(self):
+        """Orientation lines for windrose as a LineSegment2D list."""
+        lines = WindRose._xtick_radial_lines(self._bin_vectors)
+        return [line.scale(self.radius).move(self.base_point) for line in lines]
+
+    @property
+    def frequency_lines(self):
+        """Get the frequency lines for windrose as Polygon2D lists."""
+        freqs = WindRose._ytick_radial_lines(self._bin_vectors,
+                                             self.legend_parameters.segment_count)
+        return [freq.scale(self.radius).move(self.base_point) for freq in freqs]
+
+    @property
+    def radius(self):
+        """Get the radius of the inner boundary of the windrose plot.
+
+        This number will be a integer multiple of the legend parameter segement count.
+        """
+        ytick_num = self.legend_parameters.segment_count
+        if self.show_zeros:
+            ytick_num += self._ytick_zero_inc
+        return self.frequency_spacing_distance * ytick_num
+
+    @property
+    def wind_radius(self):
+        """Get the radius of the windrose mesh."""
+        if self.show_zeros:
+            return (self.real_freq_max + self.zeros_per_bin) / \
+                (self.frequency_maximum + self.zeros_per_bin) * self.radius
+        else:
+            return self.real_freq_max / self.frequency_maximum * self.radius
+
+    @property
+    def _ytick_zero_inc(self):
```

Comment:
This is minor but you don't really need to use a `@property` decorator here if the method is already hidden. The major reason why you'd use property decorators in the first place is to signify to API users that something is read-only (or you have a special setter to check the input is it isn't read-only) but, when it's already hidden, the decorator isn't really doing anything for you except saving you from an extra `()` when you call it.